microprofile-config/README.adoc

```diff
@@ -0,0 +1,793 @@
+include::../shared-doc/attributes.adoc[]
+
+= microprofile-config: MicroProfile Config QuickStart
+:author: Martin Stefanko
+:level: Beginner
+:technologies: MicroProfile Config
+
+[abstract]
+The `microprofile-config` quickstart demonstrates the use of the MicroProfile Config
+specification in {productName}.
+
+:standalone-server-type: default
+:archiveType: jar
+:archiveName: {artifactId}
+
+== What is it?
+
+MicroProfile Config allows users to externalize their configuration from their
+application code. Users can modify the configuration from outside of the application
+so they can change it without the need to rebuild their applications. It exposes the
+configuration values to the application code through the CDI injection.
+
+== Architecture
+
+In this quickstart, we have a collection of CDI beans that expose functionalities of
+the MicroProfile Config specification. The individual externally configured values
+are provided to the users through a set of REST endpoints.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the
+application step by step. However, you can go right to the completed example which
+is available in this directory.
+
+. Start your {productName} server
+
+. Compile the application code and deploy it to {productName}
+[source,options="nowrap"]
+----
+$ mvn clean package wildfly:deploy
+----
+
+// Start the {productName} Standalone Server
+include::../shared-doc/start-the-standalone-server.adoc[leveloffset=+1]
+
+== Creating the Maven Project
+
+[source,options="nowrap"]
+----
+mvn archetype:generate \
+    -DgroupId=org.wildfly.quickstart.microprofile.config \
+    -DartifactId=microprofile-config \
+    -DinteractiveMode=false \
+    -DarchetypeGroupId=org.apache.maven.archetypes \
+    -DarchetypeArtifactId=maven-archetype-webapp
```

Comment:
Just wondering, why not use the wildfly archetype?

Reply:
@rhusar is there any without Java/Jakarta mentions? I think that is why I chose to stick with default maven ones.

Reply:
If we could use any WildFly archetypes we should have merged Stefano's archetype - the problem is none of them are presently available to us once we use the quickstarts with EAP.